labs_api/insights/insights.py

```diff
@@ -56,17 +36,19 @@ def market_insights(market_id: HexAddress) -> MarketInsightsResponse:
         raise fastapi.HTTPException(
             status_code=404, detail=f"Market with id `{market_id}` not found."
         )
+    except Exception as e:
+        logger.error(f"Failed to fetch market for `{market_id}`: {e}")
+        raise fastapi.HTTPException(status_code=500)
     try:
         tavily_response = tavily_search(
             market.question_title,
             search_depth="basic",
             include_answer=True,
             max_results=5,
-            tavily_storage=TavilyStorage("market_insights"),
         )
     except Exception as e:
         logger.error(f"Failed to get tavily_response for market `{market_id}`: {e}")
-        tavily_response = None
+        raise fastapi.HTTPException(status_code=500)
```

Comment:
Don't we want to inform the user that the market wasn't found? So probably raise a 404 (or any 4xx)?

Reply:
Yes, but that is already handled above: 

![Screenshot by Dropbox Capture](https://github.com/user-attachments/assets/049a9fd4-da95-4fdb-a2b8-9f61efd10f0f)

These 500 codes should be raised only if there are unseemed errors/bugs.